contrib/tvmop/compile.py

```diff
@@ -105,18 +105,13 @@ def get_cuda_arch(arch):
         return arch.split(';')
 
     # the arch string contains '-arch=sm_xx'
+    archs = []
     flags = arch.split()
     for flag in flags:
-        if flag.startswith('-arch='):
-            return flag[len('-arch='):]
+        if flag.startswith('-gencode') or flag.startswith('arch='):
```

Comment:
`if flag.startswith('-gencode')` should never be True, as then line 104 would have been True already?

Reply:
an example input arch string would be
`-gencode arch=compute_30,code=sm_30 -gencode arch=compute_35,code=sm_35 -gencode arch=compute_50,code=sm_50 -gencode arch=compute_52,code=sm_52 -gencode arch=compute_60,code=sm_60 -gencode arch=compute_61,code=sm_61 -gencode arch=compute_70,code=sm_70 -gencode arch=compute_75,code=[sm_75,compute_75] --fatbin-options -compress-all`

L104 has semicolon ; after "gencode", I believe it's for single arch compilation @reminisce if I understand correctly.

Reply:
L104 is added for the use-case that the `arch` parameter in `get_cuda_arch(arch)` is based on cmake's `CUDA_SELECT_NVCC_ARCH_FLAGS`. But that should always have a semicolon. Is the example input arch string based on the Makefile setup?

Reply:
I believe the example comes from the Makefile setup:

https://github.com/apache/incubator-mxnet/blob/83a23b0988b55c1483f5ecec941207357492bcff/Makefile#L429-L442

Reply:
sorry I'm a bit confused, should I remove L102-L105?

Reply:
It seems CMake and make based builds pass inputs of two different formats to the `get_cuda_arch(arch)`. Before removing one condition in `get_cuda_arch(arch)`, the Makefile based build would need to be adapted to use `;` as a separator. `;` is used internally by cmake to represent lists.

For now I think it's fine to keep both, because later we anyways plan to remove the Makefile based build